projects/opendr_ws/README.md

```diff
@@ -2,65 +2,96 @@
 
 ## Description
 This ROS workspace contains ROS nodes and tools developed by OpenDR project. Currently, ROS nodes are compatible with ROS Noetic.
-This workspace contains the `ros_bridge` package, which provides message definitions for ROS-compatible OpenDR data types,
+This workspace contains the `opendr_bridge` package, which provides message definitions for ROS-compatible OpenDR data types,
 as well the `ROSBridge` class which provides an interface to convert OpenDR data types and targets into ROS-compatible
-ones similar to CvBridge. You can find more information in the corresponding [documentation](../../docs/reference/rosbridge.md).
+ones similar to CvBridge. You can find more information in the corresponding [documentation](../../docs/reference/opendr-ros-bridge.md).
 
 
-## Setup
-For running a minimal working example you can follow the instructions below:
+## First time setup
+For the initial setup you can follow the instructions below:
 
-0. Source the necessary distribution tools:
+0. Make sure ROS noetic is installed: http://wiki.ros.org/noetic/Installation/Ubuntu (desktop full install)
 
-   ```source /opt/ros/noetic/setup.bash```
+1. Open a new terminal window and source the necessary distribution tools:
+    ```shell
+    source /opt/ros/noetic/setup.bash
+    ```
+   _For convenience, you can add this line to your `.bashrc` so you don't have to source the tools each time you open a  terminal window._
+2. Install the following dependencies, required in order to use the OpenDR ROS tools:
+    ```shell
+    sudo apt-get install ros-noetic-vision-msgs ros-noetic-geometry-msgs ros-noetic-sensor-msgs ros-noetic-audio-common-msgs
+    ```
+3. Navigate to your OpenDR home directory (`~/opendr`) and activate the OpenDR environment using:
+    ```shell
+    source bin/activate.sh
+    ```
+    You need to do this step every time before running an OpenDR node.
+4. Navigate into the OpenDR ROS workspace::
+    ```shell
+    cd projects/opendr_ws
+    ```
+5. (Optional) Most nodes with visual input are set up to run with a default USB camera. If you want to use it install the corresponding package and its dependencies:
+    ```shell
+    cd src
+    git clone https://github.com/ros-drivers/usb_cam
+    cd ..
+    rosdep install --from-paths src/ --ignore-src
+    ```
+6. Build the packages inside the workspace:
+    ```shell
+    catkin_make
+    ```
+7. Source the workspace:
+    ```shell
+    source devel/setup.bash
+    ```
+   You are now ready to run an OpenDR ROS node, in this terminal but first the ROS master node needs to be running 
 
-1. Make sure you are inside opendr_ws
-2. If you are planning to use a usb camera for the demos, install the corresponding package and its dependencies:
+8. In a new terminal repeat step 1. and then run:
+    ```shell
+    roscore
```

Comment:
```suggestion
    roscore &
```

Reply:
perhaps this is better? by starting it detached there's one less terminal window you need to open

Reply:
My reasoning was to keep it as simple as possible, which indeed involves a separate terminal. I think that you are right though as it is more convenient to start it detached.

Reply:
I see, either way is fine